server/server_linux.go

```diff
@@ -139,3 +141,47 @@ func configureMaxThreads() error {
 	logrus.Debugf("Golang's threads limit set to %d", maxThreads)
 	return nil
 }
+
+// checkKernelRROMountSupport checks the kernel support for the Recursive Read-only (RRO) mounts.
+func checkKernelRROMountSupport() error {
+	// The minimal required kernel version is 5.12 or newer.
+	const kernelVersion = 512
+
+	checkKernelRROSupport := func() error {
+		var (
+			uname        unix.Utsname
+			major, minor int
+		)
+
+		if err := unix.Uname(&uname); err != nil {
+			// This should almost never happen.
+			panic(fmt.Errorf("unable to check kernel version: %w", err))
+		}
+
+		index := bytes.IndexByte(uname.Release[:], 0)
+		version := string(uname.Release[:index])
+
+		if _, err := fmt.Sscanf(version, "%d.%d", &major, &minor); err != nil {
+			return fmt.Errorf("unable to parse kernel version %q: %w", version, err)
+		}
+
+		hasMajor := major > kernelVersion/100
```

Comment:
I am fine with this for now though eventually it would be better if we found a way to check the feature without referencing the version, as it could have been backported.

Reply:
ah nevermind I see the TODO below

Reply:
yeah makes sense to me

Reply:
Done.